.github/workflows/ci.yml

```diff
@@ -0,0 +1,33 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+      - dev
+  pull_request:
+    branches:
+      - main
+      - dev
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: '22'
+
+      - name: Install dependencies
+        run: npm install
+
+      - name: Run tests
+        run: npm run test
+
```

Comment:
Test before build? Shouldn't it be the other way around?

Reply:
Yea, you're right chatGPT said test first was best from a devops perspective 

But further research suggest your suggestion:
https://www.copado.com/platform-overview?utm_source=google&utm_medium=cpc&utm_campaign=google_search_emea_en_core_cicd_nonbrand&utm_content=DevOps&utm_term=devops%20platform&gad_source=1&gbraid=0AAAAACvgvepIm88s3fMg2cNLbBORHmyxi&gclid=Cj0KCQiA_qG5BhDTARIsAA0UHSL_7ZNZe_AYQD8UnQvjt-6nnUAH5Nn1zeUU9hdprgrxCJImHrSxhxcaAuvnEALw_wcB